src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2100,22 +2112,33 @@ public RowIterator computeNext()
             }
         }
 
-        private void updateConcurrencyFactor()
+        @VisibleForTesting
+        public void handleBatchCompleted()
```

Comment:
+1 to place update of `liveReturned` here. AFIK this method doesn't do anything else than updating the concurrency factor, and we don't have any alternative implementations, so I think I'd prefer the old name for this method, `updateConcurrencyFactor`. As for the visibility change, I don't see where is it used in testing.

Reply:
renamed back to `updateConcurrencyFactor `

Reply:
I mulled around the idea of having a class whose only responsibility was tracking the command's completion state. In that case, you could imagine a method updating it in a general sense, taking the number of live rows returned and ranges queried in a round and automatically updating the concurrency factor (which you would just access to make the next batch). It would be pretty easy to test and make `RangeCommandIterator` a little more focused, but the static `computeConcurrencyFactor()` basically gets us the same thing.

tl;dr I have no problems with the current structure.